packages/auth/amplify_auth_cognito/lib/src/auth_plugin_impl.dart

```diff
@@ -120,6 +120,20 @@ class AmplifyAuthCognito extends AmplifyAuthCognitoDart with AWSDebuggable {
     AmplifyConfig? config,
     required AmplifyAuthProviderRepository authProviderRepo,
   }) async {
+    // Dependencies for AnalyticsMetadataType
```

Comment:
I added the state machine dependencies here because the configuration state machine is called during the call to configure in dart. 

I don't think this code is ideal as I'm relying on the execution order of the code to get things to work. 

@dnys1 can I move this code to the constructor instead?  I'm a bit concerned these changes are going to make the code less maintainable in the future.